src/telemetry.c

```diff
@@ -1016,6 +1090,55 @@ int tm_set_payload(struct telem_ref *t_ref, char *payload)
         return ret;
 }
 
+
+/**
+ * Checks for id to have length = 32 characters
+ * and hexadecimal characters only.
+ *
+ * @param id A pointer to string to be checked
+ *
+ * @return 0 on success, 1 on failure
+ *
+ */
+static int validate_event_id(char *id)
```

Comment:
Should this function be in a common place so it can be reused from telem_record_gen.c?